Cannot take backup to Azure when using managed identity

**Describe the bug**

In `8.5.1` Zeebe cannot take backups to Azure storage when using [managed identity](https://learn.microsoft.com/en-us/azure/aks/use-managed-identity) for Azure. It works in `8.5.0`

```
io.camunda.zeebe.backup.management.BackupService - Failed to take backup 110 at position 1
java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: 'com.microsoft.aad.msal4j.AbstractApplicationBase$Builder com.microsoft.aad.msal4j.ConfidentialClientApplication$Builder.logPii(boolean)'
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(Unknown Source) ~[?:?]
	at java.base/java.lang.VirtualThread.run(Unknown Source) ~[?:?]
Caused by: java.lang.NoSuchMethodError: 'com.microsoft.aad.msal4j.AbstractApplicationBase$Builder com.microsoft.aad.msal4j.ConfidentialClientApplication$Builder.logPii(boolean)'
	at com.azure.identity.implementation.IdentityClientBase.getWorkloadIdentityConfidentialClient(IdentityClientBase.java:455) ~[azure-identity-1.12.0.jar:1.12.0]
	at com.azure.identity.implementation.IdentityClient.lambda$getWorkloadIdentityConfidentialClientApplication$6(IdentityClient.java:150) ~[azure-identity-1.12.0.jar:1.12.0]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:45) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoCacheTime.subscribeOrReturn(MonoCacheTime.java:143) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4552) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:430) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxIterable$IterableSubscription.slowPath(FluxIterable.java:335) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxIterable$IterableSubscription.request(FluxIterable.java:294) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onSubscribe(FluxFlatMap.java:373) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:201) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:83) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoUsing.subscribe(MonoUsing.java:102) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoFromFluxOperator.subscribe(MonoFromFluxOperator.java:83) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.subscribeNext(MonoIgnoreThen.java:265) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:51) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:205) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:63) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:205) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.subscribe(Mono.java:4552) ~[reactor-core-3.6.5.jar:3.6.5]
	at reactor.core.publisher.Mono.block(Mono.java:1778) ~[reactor-core-3.6.5.jar:3.6.5]
	at com.azure.storage.common.implementation.StorageImplUtils.blockWithOptionalTimeout(StorageImplUtils.java:147) ~[azure-storage-common-12.24.4.jar:12.24.4]
	at com.azure.storage.blob.BlobContainerClient.createIfNotExistsWithResponse(BlobContainerClient.java:374) ~[azure-storage-blob-12.25.4.jar:12.25.4]
	at com.azure.storage.blob.BlobContainerClient.createIfNotExists(BlobContainerClient.java:340) ~[azure-storage-blob-12.25.4.jar:12.25.4]
	at io.camunda.zeebe.backup.azure.ManifestManager.assureContainerCreated(ManifestManager.java:260) ~[zeebe-backup-store-azure-8.5.1.jar:8.5.1]
	at io.camunda.zeebe.backup.azure.ManifestManager.listManifests(ManifestManager.java:213) ~[zeebe-backup-store-azure-8.5.1.jar:8.5.1]
	at io.camunda.zeebe.backup.azure.AzureBackupStore.lambda$list$2(AzureBackupStore.java:128) ~[zeebe-backup-store-azure-8.5.1.jar:8.5.1]
	... 3 more
```

The root cause is a mismatch in the versions between `azure-identity` and `msal4j`. We are pinning `msal4j` to `1.14.0`, but azure-identity needs `1.15.0`. 

Workaround is to configure access-key via environment variables. But that is not the preferred way for many users.

**Environment:**
- Zeebe Version: 8.5.1, 8.5.2
- Configuration: Azure backup storage with managed identity

related to https://jira.camunda.com/browse/SUPPORT-22209

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -141,7 +141,6 @@
     <version.jakarta.json-api>2.1.3</version.jakarta.json-api>
     <version.jakarta-annotation>3.0.0</version.jakarta-annotation>
     <version.jna-platform>5.14.0</version.jna-platform>
-    <version.msal4j>1.15.1</version.msal4j>
 
     <version.commons-io>2.16.1</version.commons-io>
     <version.thymeleaf>3.1.2.RELEASE</version.thymeleaf>
@@ -1753,13 +1752,6 @@
         <groupId>jakarta.json</groupId>
         <artifactId>jakarta.json-api</artifactId>
         <version>${version.jakarta.json-api}</version>
-      </dependency>
-
-      <!-- internal conflicts from azure-identity -->
-      <dependency>
-        <groupId>com.microsoft.azure</groupId>
-        <artifactId>msal4j</artifactId>
-        <version>${version.msal4j}</version>
       </dependency>
 
       <dependency>

